testing/python/fixtures.py

```diff
@@ -121,16 +121,20 @@ def test_detect_recursive_dependency_error(self, pytester: Pytester) -> None:
             ["*recursive dependency involving fixture 'fix1' detected*"]
         )
 
-    def test_funcarg_basic(self, pytester: Pytester) -> None:
+    def test_funcarg_basic(self, recwarn, pytester: Pytester) -> None:
         pytester.copy_example()
         item = pytester.getitem(Path("test_funcarg_basic.py"))
         assert isinstance(item, Function)
         # Execute's item's setup, which fills fixtures.
         item.session._setupstate.setup(item)
+        assert len(recwarn) == 0
```

Comment:
It will be better to move this test (only the deprecation part) into `testing/deprecated_test.py` for easy removal later on.